### ISSUE TITLE
TypeError: cannot read property 0 of undefined of cannot read property slice of undefined

### ISSUE THREAD

**Issue by xxhenglyxx** [brix/crypto-js#156]
I warp your library to create my custom encrypt/decrypt function, here is what it looks like:

    import Random from 'random-js';
    import randomBytes from 'randombytes';
    import CryptoJS from 'crypto-js';

    const { TripleDES, AES } = CryptoJS;
    const encryptionMethod = [ TripleDES, AES ];
    const random = new Random ();

    export const encryptJSON = function ( data ) {

        const secret = randomBytes ( 68 );
        const method = random.integer ( 0, 1 );

        const encrypted = encryptionMethod [ method ].encrypt ( JSON.stringify ( data ), secret );

        return [ encrypted, method, secret ];

    };

    export const decryptJSON = function ( encrypted, method, secret ) {

        return JSON.parse ( ( encryptionMethod [ method ].decrypt ( encrypted.toString (), secret ) ).toString ( CryptoJS.enc.Utf8 ) );

    };

I am testing it with simple object pass in as arguments, bind the function to a button to test it

    import ...

    class SimpleCryptoTest extends React.PureComponent {

        testCrypto = () => {

            let data = encryptJSON ( { t: 123, m: "adfasd" } );

            console.log ( data );

            data = decryptJSON ( ...data );

            console.log ( data );

        }

        ...

    }

Here is the console outputs

![screen shot 2018-04-05 at 4 15 26 pm](https://user-images.githubusercontent.com/28291947/38357363-975a893e-38ec-11e8-8f17-a37dece7dff7.png)

The order of the error sometime random, it outputs the cannot read 0 first sometime or cannot read property slice sometime

**Comment by sorenrife** [comment]
Same here

**Comment by dpc11** [comment]
Said it was because the message or the key should be string, still not works for me:
https://stackoverflow.com/questions/50979762/uncaught-typeerror-cannot-read-property-0-of-undefined-cryptojs

**Comment by dpc11** [comment]
Solved in my case by changing the encrypt mode to `CryptoJS.mode.ECB` or setting `iv` parameter for the default `CryptoJS.mode.CBC`,

The error itself didn't provide any information, it can be caused by any parameter error I guess.

**Comment by Alanscut** [comment]
> Said it was because the message or the key should be string, still not works for me:
> https://stackoverflow.com/questions/50979762/uncaught-typeerror-cannot-read-property-0-of-undefined-cryptojs

@xxhenglyxx 
Yes, the key should be string.You can try `const secret = randomBytes ( 68 ).toString('utf8');`
